qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,522 @@
+        full_sampling: bool = False,
+        inverting_pauli_layer: bool = False,
+    ):
+        """Initialize a mirror randomized benchmarking experiment that uses
+        pyGSTi's circuit generation.
+
+        Args:
+            qubits: A list of physical qubits for the experiment.
+            lengths: A list of RB sequences lengths.
+            local_clifford: If True, begin the circuit with uniformly random 1-qubit
+                            Cliffords and end the circuit with their inverses.
+            pauli_randomize: If True, surround each inner Clifford layer with
+                            uniformly random Paulis.
+            two_qubit_gate_density: Expected proportion of qubits with CNOTs based on
+                                    the backend coupling map.
+            backend: The backend to run the experiment on.
+            num_samples: Number of samples to generate for each
+                        sequence length.
+            seed: Optional, seed used to initialize ``numpy.random.default_rng``.
+                when generating circuits. The ``default_rng`` will be initialized
+                with this seed value everytime :meth:`circuits` is called.
+            full_sampling: If True all Cliffords are independently sampled for
+                        all lengths. If False for sample of lengths longer
+                        sequences are constructed by appending additional
+                        Clifford samples to shorter sequences.
+            inverting_pauli_layer: If True, a layer of Pauli gates is appended at the
+                                end of the circuit to set all qubits to 0 (with
+                                possibly a global phase)
+
+        Raises:
+            ImportError: if user does not have pyGSTi installed
+        """
+        if not HAS_PYGSTI:
+            raise ImportError("MirrorRBPyGSTi requires pyGSTi to generate circuits.")
+
+        super().__init__(
+            qubits,
+            lengths,
+            local_clifford=local_clifford,
+            pauli_randomize=pauli_randomize,
+            two_qubit_gate_density=two_qubit_gate_density,
+            backend=backend,
+            num_samples=num_samples,
+            seed=seed,
+            full_sampling=full_sampling,
+            inverting_pauli_layer=inverting_pauli_layer,
+        )
+
+        self.analysis = MirrorRBAnalysis()
+        self._lengths = lengths
+        self._num_samples = num_samples
+        self._seed = seed
+        self.analysis.set_options(outcome="0" * self.num_qubits)
+
+    def _transpiled_circuits(self) -> List[QuantumCircuit]:
+        """Return a list of experiment circuits, transpiled, with transpiled
+        circuits stored as metadata."""
+        transpiled = super()._transpiled_circuits()
+
+        # Store transpiled circuits in metadata
+        for circ in transpiled:
+            circ.metadata["transpiled_qiskit_circ"] = circ
```

Comment:
This will break QPY serialization I guess unless you provide custom Json serializer.

Reply:
We've moved this to a PR in the internal repository